### PR TITLE
Add launchDataChanged message on Android

### PR DESCRIPTION
**Pull request by livecodeian**

### CODE REVIEW

File: engine/src/mblandroiddc.cpp [hunk 1]
```diff
@@ -2229,6 +2229,16 @@ JNIEXPORT void JNICALL Java_com_runrev_android_Engine_doLaunchFromUrl(JNIEnv *en
...
         MCNotificationPostUrlWakeUp(*t_url_str);
 }
+
+extern "C" JNIEXPORT void JNICALL Java_com_runrev_android_Engine_doLaunchDataChanged(JNIEnv *env, jobject object, jobject launch_data) __attribute__((visibility("default")));
+JNIEXPORT void JNICALL Java_com_runrev_android_Engine_doLaunchDataChanged(JNIEnv *env, jobject object, jobject launch_data)
+{
+	
+	MCAutoArrayRef t_array;
+	if (MCJavaMapToArrayRef(env, launch_data, &t_array))
+		/* UNCHECKED */ MCNotificationPostLaunchDataChanged(*t_array);
```
Comment (runrevmark) [inline engine/src/mblandroiddc.cpp]: Do we need to pass a parameter to the notification? After all the new launch data is available via the standard function.

**Comment by livecodeian** [comment]
@runrevmark It's consistent with "urlWakeUp pNewUrl" but I can remove it if you need me to.

**Comment by runrevmark** [comment]
@livecodeian: I think it is redundant - it encourages people to think they have to take the intent from that parameter, rather than just using the global environment function. I'd argue the same is true of the urlWakeUp message (assuming there's a global function to get that messages parameter too).

File: docs/dictionary/message/launchDataChanged.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    <mobile/>
+  </classes>
+  <security>
+  </security>
+  <description>
+    <overview>Sent to the <glossary tag="current card">current card</glossary> when the <glossary tag="application">application</glossary>'s launch data is changed by a new request from another app.</overview>
+    <comments>An Android app may be suspended in the background when it receives a request from another app. When this happens, the app's launch data is updated to contain the data for the new request and the app is resumed. The new launch data can be obtained from the <function tag="mobileGetLaunchData">mobileGetLaunchData</function> function.</comments>
+  </description>
+  <examples>
+    <example><p>on launchDataChanged pURL</p><p>   myLaunchHandler mobileGetLaunchData()</p><p>end launchDataChanged</p></example>
```
Comment (runrevmark) [inline docs/dictionary/message/launchDataChanged.xml]: This needs the pURL parameter removed :)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok ed38a48

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok ed38a48ebbc9f2af405769b13fa379abf05e6ce7

**Comment by livecode-vulcan** [comment]
😎 test success ed38a48ebbc9f2af405769b13fa379abf05e6ce7
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/712)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/725)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/713)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/715)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/725)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/736)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/727)

**Pull request merged**